mvpa2/cmdline/cmd_plotmotionqc.py

```diff
@@ -0,0 +1,159 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Generate a QC plot for (BOLD fMRI) motion estimates of multiple segments.
+
+The generated figure consists of two subplots: one for translation and one for
+rotation. The L2-norm for each motion type is plotted. Segment boundaries
+are indicated with dashed vertical lines. The following statistics are
+visualized
+
+1. Range across subjects (min, max) with a light gray shaded area
+
+2. 50% percentile around the mean with a medium gray shaded area
+
+3. Standard error of the mean (SEM; +/-) with a dark gray shaded area
+
+4. Median across subjects with a black line
+
+5. Outlier subjects are represented as individual red lines
+
+Outliers are defined as subjects that exceed the threshold anywhere within
+a given segment. In that case the entire motion time series for that segment is
+plotted as an outlier.
+
+Example
+
+pymvpa2 plotmotionqc \
+    -s sub*/func/*run-1_bold_mc.txt \
+    -s sub*/func/*run-2_bold_mc.txt \
+    --savefig motion.png
+
+"""
+
+# magic line for manpage summary
+# man: -*- % BOLD fMRI motion QC plot
+
+__docformat__ = 'restructuredtext'
+
+import argparse
+import numpy as np
+
+parser_args = {
+    'formatter_class': argparse.RawDescriptionHelpFormatter,
+}
+
+
+def setup_parser(parser):
+    parser.add_argument(
+        '-s', '--segment', metavar='FILE', type=np.loadtxt, nargs='+',
+        action='append',
+        help="""Two or more text files with motion estimate time series.
+        This option can be given multiple times (with multiple time series
+        each to generate a multi-segment plot (e.g. for multiple run).""")
+    parser.add_argument(
+        '--estimate-order', metavar='LABEL', default='transrot',
+        choices=('transrot', 'rottrans'),
+        help="""column order of estimates in the files. `transrot` indicates
+        translation first, followed by rotation. `rottrans` refers to the
+        oposite order. Default: 'transrot'""")
```

Comment:
param2arg added docs use `[Default: %r]` so would be better to keep it uniform.
But I wonder if it would better be possible even to automate it to avoid manually listing default value... iirc argparse allows that somehow... https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter  ?